### PR TITLE
Feature/use be configs

### PR DESCRIPTION
**Pull request by mloeffle**
- add pinia store for config values
- get function for config entries
- example usage by implementing toggle for wishlist

### CODE REVIEW

File: components/Product/ProductCard.vue [hunk 1]
```diff
@@ -15,11 +15,16 @@ const props = withDefaults(
...
+const configStore = useConfigStore();
+const wishlistEnabled = configStore.get('core.cart.wishlistEnabled');
 </script>
 
 <template>
     <div class="relative rounded-md p-4 shadow-md">
-        <div class="absolute right-0 top-0 z-[2] p-4">
+        <div
+            v-if="wishlistEnabled"
+            class="absolute right-0 top-0 z-[2] p-4"
```
Comment (aheartforspinach) [inline components/Product/ProductCard.vue:26]: I know that you wasn't the person who add the `z-[2]` but could we replace it with a default tailwind class?

File: composables/usePluginConfig.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export function usePluginConfig() {
+    const { apiClient } = useShopwareContext();
+    const configValues = ref({});
+
+    const loadConfig = async () => {
+        return useAsyncData('pluginConfiguration', async () => {
+            const data = (await apiClient.invoke('loadConfig get /pond/config')).data;
```
Comment (aheartforspinach) [inline composables/usePluginConfig.ts:7]: we should add a try catch for the case when something is not status code 200

File: stores/ConfigStore.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export const useConfigStore = defineStore('config', () => {
+    const { loadConfig, configValues } = usePluginConfig();
+
+    const get = (key: string) => {
+        if (!configValues.value) {
+            console.warn('[Pond]: config values not loaded');
```
Comment (aheartforspinach) [inline stores/ConfigStore.ts:6]: could we only throw these warnings in development?

File: stores/ConfigStore.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            console.warn(`[Pond]: The configuration for '${key}' was not found`);
+            return undefined;
+        }
+
+        return configValues.value[key];
+    };
+
+    return {
+        loadConfig,
+        configValues,
```
Comment (aheartforspinach) [inline stores/ConfigStore.ts:20]: question: why should we allow an user to load the config again or get access directly to the config values? wouldn't it be enough to just return the get function?

File: components/Layout/Header/LayoutHeaderWishlist.vue [hunk 5]
```diff
@@ -1,9 +1,12 @@
 <script setup lang="ts">
 const { count: wishlistCount } = useWishlist();
+const configStore = useConfigStore();
+const wishlistEnabled = configStore.get('core.cart.wishlistEnabled');
 </script>
 
 <template>
     <NuxtLink
+        v-if="wishlistEnabled"
```
Comment (m-borgmann) [inline components/Layout/Header/LayoutHeaderWishlist.vue:9]: The check should be done for the entire component and not just the link. So a better position would be in the LayoutHeaderActions file. 

I also just merged some fixes for the wishlist which moved the call for wishlist products to the layout/default.vue file. That call should also only be done when the wishlist is enabled. It should be easy to add the check for that there aswell

**Review by m-borgmann — APPROVED** [review]

File: components/Layout/Header/LayoutHeaderActions.vue [hunk 6]
```diff
@@ -1,8 +1,12 @@
+<script setup lang="ts">
+const wishlistEnabled = configStore.get('core.cart.wishlistEnabled');
```
Comment (aheartforspinach) [inline components/Layout/Header/LayoutHeaderActions.vue:2]: Isnt `const configStore = useConfigStore();` missing here?

File: components/Product/ProductCard.vue [hunk 7]
```diff
@@ -15,11 +15,17 @@ const props = withDefaults(
 const { getProductCover } = useMedia();
 
 const cover = getProductCover(props.product.cover);
+const configStore = useConfigStore();
+const wishlistEnabled = configStore.get('core.cart.wishlistEnabled');
+configStore.get('test');
```
Comment (aheartforspinach) [inline components/Product/ProductCard.vue:20]: why test?

File: composables/usePluginConfig.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    const { apiClient } = useShopwareContext();
+    const configValues = ref({});
+
+    const loadConfig = async () => {
+        return useAsyncData('pluginConfiguration', async () => {
+            try {
+                const data = (await apiClient.invoke('loadConfig get /pond/config')).data;
+                configValues.value = data;
+                return data;
+            } catch (e) {
```
Comment (aheartforspinach) [inline composables/usePluginConfig.ts:11]:
```suggestion
            } catch (error) {
```

**Review by aheartforspinach — APPROVED** [review]

**Pull request merged**